### PR TITLE
Support Gradle 8.0

### LINKED ISSUE #4: Migrate convention to extension
The concept of conventions is outdated and superseded by extensions. 

https://docs.gradle.org/current/userguide/upgrading_version_7.html#all_convention_deprecation

### LINKED ISSUE #5: Support Gradle 8.0
Links #4.

### PR DESCRIPTION
**Pull request by Goooler**
- Closes #4.
- Closes #5.

### CODE REVIEW

File: src/main/resources/META-INF/gradle-plugins/com.github.blindpirate.osgi.properties [hunk 1]
```diff
@@ -1 +0,0 @@
-implementation-class=com.github.blindpirate.osgi.plugins.osgi.OsgiPlugin
```
Comment (blindpirate) [inline src/main/resources/META-INF/gradle-plugins/com.github.blindpirate.osgi.properties:1]: Why is this removed?

**Review by blindpirate — COMMENTED** [review]
This is so great, really appreciate it! There is only one question, `src/main/resources/META-INF/gradle-plugins/com.github.blindpirate.osgi.properties` is used as the metadata of the plugin so I think it's necessary. Why it's removed?

File: src/main/resources/META-INF/gradle-plugins/com.github.blindpirate.osgi.properties (hunk 1 above)
Comment (Goooler) [reply src/main/resources/META-INF/gradle-plugins/com.github.blindpirate.osgi.properties:1]: It was replaced by [implementationClass](https://github.com/Goooler/gradle-legacy-osgi-plugin/blob/60dfd00066c2b4cdc267c7c5c5b74dc2d5254534/build.gradle#L37).

Comment (Goooler) [reply src/main/resources/META-INF/gradle-plugins/com.github.blindpirate.osgi.properties:1]: See https://docs.gradle.org/current/userguide/java_gradle_plugin.html#sec:gradle_plugin_dev_usage

Comment (blindpirate) [reply src/main/resources/META-INF/gradle-plugins/com.github.blindpirate.osgi.properties:1]: OK, I see the `properties` exists in the generated jar - it might be synthesized. I was somewhat misled by the warning:

```
> Task :jar
:jar: No valid plugin descriptors were found in META-INF/gradle-plugins
```

Anyway, loos good!

**Comment by blindpirate** [comment]
`0.0.7` has been published but it was somehow not displayed on website yet. Will follow up on this.

**Comment by Goooler** [comment]
Due to the coordinate id change, see https://github.com/johnrengelman/shadow/issues/827#issuecomment-1446667743.

**Comment by Goooler** [comment]
One more thing, should we enable @renovate-bot for this repo?

**Comment by Goooler** [comment]
It's out, check https://plugins.gradle.org/plugin/com.github.blindpirate.osgi
